### ISSUE TITLE
quarto publish frozen at push, web page 404's. 

### ISSUE THREAD

**Issue by drscotthawley** [quarto-dev/quarto-cli#3368]
### Bug description

New user here, migrating from fastpages. 

I ran `quarto publish` after [migrating my fastpages blog](https://nbdev.fast.ai/tutorials/blogging.html#migrating-from-fastpages), and selected the GitHub Pages deployment option. Despite the [GitHub deployment workflow](https://github.com/drscotthawley/blog_quarto/actions/runs/3494794082)  being completed, `quarto publish` seems to be frozen at its final "push":   

No CPU usage (from `quarto` or `deno` or `git`)  is happening, and there' no outbound network traffic.  A look at the [`gh-pages` branch of my repo](https://github.com/drscotthawley/blog_quarto/tree/gh-pages) shows it to be empty. 

There is a lot of material to this first commit/publish, however I have waited probably half an hour now, and like I say there is neither CPU activity nor network activity. 

Here is a list of the output log: 
```
$ quarto publish
? Provider: › GitHub Pages
? Publish site to https://drscotthawley.github.io/blog_quarto/ using gh-pages? (Y/n) › Yes
Switched to a new branch 'gh-pages'
[gh-pages (root-commit) 98a3e04] Initializing gh-pages branch
remote: 
remote: Create a pull request for 'gh-pages' on GitHub by visiting:        
remote:      https://github.com/drscotthawley/blog_quarto/pull/new/gh-pages        
remote: 
To https://github.com/drscotthawley/blog_quarto.git
 * [new branch]      HEAD -> gh-pages
Your branch is up to date with 'origin/main'.
Switched to branch 'main'
From https://github.com/drscotthawley/blog_quarto
 * branch            gh-pages   -> FETCH_HEAD
Rendering for publish:

[ 1/35] posts/2020-03-26-Sliding-Off-A-Hill.ipynb
[ 2/35] posts/2017-02-03-How-To-Port-Forward-Jupyter-Notebooks.md
[ 3/35] posts/post-with-code/index.qmd
[ 4/35] posts/2019-02-08-My-1st-NN-Part-3-Multi-Layer-and-Backprop.ipynb
[ 5/35] posts/2021-06-17-Contrasting-Contrastive-Loss.ipynb
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"6256f42ad778484b84c4d61aa8fbddaa","version_major":2,"version_minor":0}
[ 6/35] posts/2020-01-13-Mac-Linux-Switch-Notes.md
[ 7/35] posts/2021-05-04-The-Joy-Of-3D.ipynb
[ 8/35] posts/2021-09-15-Handy-Bash-Functions.md
[ 9/35] posts/2020-11-01-Interactivity_Test.md
[10/35] posts/2022-11-15-MultiDim-Shells.ipynb
[11/35] posts/2022-01-24-MultiDim-DotProducts.ipynb
[12/35] posts/2018-02-21-Resolving-OSX-Aliases.md
[13/35] posts/2019-01-30-My-First-Neural-Network.ipynb
[14/35] posts/2020-03-01-Data-Are-Measurements.ipynb
[15/35] posts/2017-05-05-Crash-Course-On-GANs.md
[16/35] posts/welcome/index.qmd
[17/35] posts/2021-02-15-Making-GPT3-Generate-3-Of-A-Kind.md
[18/35] posts/2020-07-01-Citations-Via-Bibtex.md
[19/35] posts/2022-11-17-BYOL.ipynb
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"0a13f79dfac44ab4bd5eb9cde8825cfe","version_major":2,"version_minor":0}
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"104c60786f344b0fa79c7ac973a282b7","version_major":2,"version_minor":0}
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"b0c6c1e53d7a43e097b27eefec24ca97","version_major":2,"version_minor":0}
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"bb1fb4af6088404ba78802e1a271dee9","version_major":2,"version_minor":0}
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"c2798d977d8c4e398e03df92729388c8","version_major":2,"version_minor":0}
WARNING: Malformed Jupyter Output Display Data found:
WARNING: {"model_id":"20344289a17c4a05991fa221d2954a13","version_major":2,"version_minor":0}
[20/35] posts/2022-02-15-Data-Identity.md
[21/35] posts/2020-12-11-Naughty-by-Numbers.md
[22/35] posts/2018-12-16-Parallelizing-Python-Simplified.md
[23/35] posts/2021-08-06-Live-CL-Demo.md
[24/35] posts/2021-10-15-Geometric-Algebra.ipynb
[25/35] posts/2019-02-04-My-First-NN-Part-2.ipynb
[26/35] posts/2017-11-22-Convexity-Loss-Classification.ipynb
[27/35] posts/2021-05-04-The-Joy-Of-3D-OLD.ipynb
[28/35] posts/2019-12-06-Getting-Started-In-ML-Audio.md
[29/35] posts/2017-05-04-Learning-Room-Shapes.md
[30/35] posts/2020-01-14-test-markdown-post.md
[31/35] posts/2017-02-23-Following-Gravity-Colab.ipynb
[32/35] posts/2017-02-06-Machine-Learning-Reference-List.md
[33/35] posts/2019-12-21-PCA-From-Scratch.ipynb
[34/35] index.qmd
[35/35] about.qmd

Preparing worktree (resetting branch 'gh-pages'; was at 98a3e04)
Branch 'gh-pages' set up to track remote branch 'gh-pages' from 'origin'.
HEAD is now at 98a3e04 Initializing gh-pages branch
[gh-pages 5324814] Built site for gh-pages
 206 files changed, 34716 insertions(+)
 create mode 100644 .nojekyll
 create mode 100644 2017/02/03/How-To-Port-Forward-Jupyter-Notebooks/index.html
 create mode 100644 2017/02/06/Machine-Learning-Reference-List/index.html
 create mode 100644 2017/02/23/Following-Gravity-Colab/index.html
 create mode 100644 2017/05/04/Learning-Room-Shapes/index.html
 create mode 100644 2017/05/05/Crash-Course-On-GANs/index.html
 create mode 100644 2017/11/22/Convexity-Loss-Classification/index.html
 create mode 100644 2018/02/21/Resolving-OSX-Aliases/index.html
 create mode 100644 2018/12/16/Parallelizing-Python-Simplified/index.html
 create mode 100644 2019/01/30/My-First-Neural-Network/index.html
 create mode 100644 2019/02/04/My-First-NN-Part-2/index.html
 create mode 100644 2019/02/08/My-1st-NN-Part-3-Multi-Layer-and-Backprop/index.html
 create mode 100644 2019/12/06/Getting-Started-In-ML-Audio/index.html
 create mode 100644 2019/12/21/PCA-From-Scratch/index.html
 create mode 100644 2020/01/13/Mac-Linux-Switch-Notes/index.html
 create mode 100644 2020/07/01/Citations-Via-Bibtex/index.html
 create mode 100644 2020/11/01/Interactivity_Test/index.html
 create mode 100644 2020/12/11/Naughty-by-Numbers/index.html
 create mode 100644 2021/08/06/Live-CL-Demo/index.html
 create mode 100644 2021/09/15/Handy-Bash-Functions/index.html
 create mode 100644 2021/10/15/Geometric-Algebra/index.html
 create mode 100644 2022/01/24/MultiDim-DotProducts/index.html
 create mode 100644 2022/02/15/Data-Identity/index.html
 create mode 100644 2022/11/15/MultiDim-Shells/index.html
 create mode 100644 2022/11/17/BYOL/index.html
 create mode 100644 about.html
 create mode 100644 classification/gpt-3/scottergories/2021/02/15/Making-GPT3-Generate-3-Of-A-Kind/index.html
 create mode 100644 coronavirus/physics/2020/03/26/Sliding-Off-A-Hill/index.html
 create mode 100644 data/deepfakes/philosophy/2020/03/01/Data-Are-Measurements/index.html
 create mode 100644 index.html
 create mode 100644 listings.json
 create mode 100644 markdown/2020/01/14/test-markdown-post/index.html
 create mode 100644 posts/2017-02-03-How-To-Port-Forward-Jupyter-Notebooks.html
 create mode 100644 posts/2017-02-06-Machine-Learning-Reference-List.html
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab.html
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-10-output-1.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-11-output-1.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-12-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-13-output-1.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-14-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-15-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-16-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-2-output-1.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-3-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-4-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-5-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-7-output-2.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-8-output-1.png
 create mode 100644 posts/2017-02-23-Following-Gravity-Colab_files/figure-html/cell-9-output-2.png
 create mode 100644 posts/2017-05-04-Learning-Room-Shapes.html
 create mode 100644 posts/2017-05-05-Crash-Course-On-GANs.html
 create mode 100644 posts/2017-11-22-Convexity-Loss-Classification.html
 create mode 100644 posts/2017-11-22-Convexity-Loss-Classification_files/figure-html/cell-4-output-2.png
 create mode 100644 posts/2017-11-22-Convexity-Loss-Classification_files/figure-html/cell-6-output-1.png
 create mode 100644 posts/2017-11-22-Convexity-Loss-Classification_files/figure-html/cell-7-output-1.png
 create mode 100644 posts/2017-11-22-Convexity-Loss-Classification_files/figure-html/cell-7-output-2.png
 create mode 100644 posts/2018-02-21-Resolving-OSX-Aliases.html
 create mode 100644 posts/2018-12-16-Parallelizing-Python-Simplified.html
 create mode 100644 posts/2019-01-30-My-First-Neural-Network.html
 create mode 100644 posts/2019-01-30-My-First-Neural-Network_files/figure-html/cell-3-output-1.png
 create mode 100644 posts/2019-01-30-My-First-Neural-Network_files/figure-html/cell-6-output-1.png
 create mode 100644 posts/2019-02-04-My-First-NN-Part-2.html
 create mode 100644 posts/2019-02-04-My-First-NN-Part-2_files/figure-html/cell-3-output-1.png
 create mode 100644 posts/2019-02-04-My-First-NN-Part-2_files/figure-html/cell-4-output-2.png
 create mode 100644 posts/2019-02-04-My-First-NN-Part-2_files/figure-html/cell-4-output-4.png
 create mode 100644 posts/2019-02-04-My-First-NN-Part-2_files/figure-html/cell-6-output-1.png
 create mode 100644 posts/2019-02-04-My-First-NN-Part-2_files/figure-html/cell-7-output-2.png
 create mode 100644 posts/2019-02-08-My-1st-NN-Part-3-Multi-Layer-and-Backprop_files/figure-html/cell-4-output-1.png
 create mode 100644 posts/2019-02-08-My-1st-NN-Part-3-Multi-Layer-and-Backprop_files/figure-html/cell-5-output-2.png
 create mode 100644 posts/2019-02-08-My-1st-NN-Part-3-Multi-Layer-and-Backprop_files/figure-html/cell-6-output-2.png
 create mode 100644 posts/2019-02-08-My-1st-NN-Part-3-Multi-Layer-and-Backprop_files/figure-html/cell-8-output-2.png
 create mode 100644 posts/2019-02-08-My-1st-NN-Part-3-Multi-Layer-and-Backprop_files/figure-html/cell-9-output-2.png
 create mode 100644 posts/2019-02-08-my-1st-nn-part-3-multi-layer-and-backprop.html
 create mode 100644 posts/2019-12-06-Getting-Started-In-ML-Audio.html
 create mode 100644 posts/2019-12-21-PCA-From-Scratch.horigin	https://github.com/drscotthawley/blog_quarto.git (fetch)
origin	https://github.com/drscotthawley/blog_quarto.git (push)


<frozen for 20 minutes> 
```

"Please include a minimal, fully reproducible example in a single .qmd file?"  Sorry I don't know what a .qmd file is.  But I've included a full output log above.

No using RStudio IDE. 

Running on Mac Os Monterey, 12.3

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [X] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by drscotthawley** [comment]
If I use a different Terminal window and execute a `git checkout gh-pages` I'm told that `gh-pages` branch already checked out as directory `e5413c3f/`,  which seems to contain all the sorts of content one might want from a blog: 
```bash
$ ls e5413c3f/
2017/           2020/           about.html      data/           markdown/       scottergories/  styles.css
2018/           2021/           classification/ index.html      posts/          search.json
2019/           2022/           coronavirus/    listings.json   profile.jpg     site_libs/
```

I have just executed a manual `git push` from within that directory  (which may be clobbering an existing push from the stalled `quarto publish` op but at this point I"m not sure what to do).  Here's what I see from that: 

```bash
$ cd e5413c3f/
$ git push 
Enumerating objects: 339, done.
Counting objects: 100% (339/339), done.
Delta compression using up to 10 threads
Compressing objects: 100% (239/239), done.
Writing objects: 100% (338/338), 18.94 MiB | 1.31 MiB/s, done.
Total 338 (delta 48), reused 0 (delta 0), pack-reused 0
```
...and then that seems to hang too.  Perhaps this is a temporary GitHub issue?

**Comment by drscotthawley** [comment]
(Both processes) Still hung the next morning (7 hours later).  Going to abort via ^C and retry `quarto publish`...

**Comment by drscotthawley** [comment]
Ok this time I get a bit more output and something seems to be happening.  Generally same output as above, but then it goes on to say...

```bash
To https://github.com/drscotthawley/blog_quarto.git
 + 98a3e04...14c18b5 HEAD -> gh-pages (forced update)
fatal: '42b19423' is not a working tree

NOTE: GitHub Pages sites use caching so you might need to click the refresh
button within your web browser to see changes after deployment.

(-) Deploying gh-pages branch to website (this may take a few minutes)
```
Happy to wait a few minutes... and report back 🤞

**Comment by drscotthawley** [comment]
....Yikes, just noticed that "fatal" message above.   Ok, let's got back and check that I'm really on `main` before I run `quarto publish` (and didn't accidentally try to run `quarto publish` while `gh-pages` is checked out) ...  and now re-running...

😭 
Back to the original hanging behavior, immediately after "origin...push".  No output after that, i.e. no "this may take a few minutes"   

Will love to hear suggestions at this point, I'm not sure what else to try. 🤷

**Comment by cscheid** [comment]
Is your git repository on a non-standard filesystem, by any chance? eg a dropbox folder, google drive, onedrive

**Comment by drscotthawley** [comment]
Tried deleting all the various subdirectories that had formed at this point: 

```bash
$ rm -rf 42b19423 55c0c37e 70effac0/ a36d440f/ e5413c3f/
```

then

```bash
$ quarto publish
...
...
origin https://github.com/drscotthawley/blog_quarto.git (push)
[nothing more]
```

**Comment by drscotthawley** [comment]
> Is your git repository on a non-standard filesystem, by any chance? eg a dropbox folder, google drive, onedrive

No, just default Mac filesystem.

**Comment by cscheid** [comment]
Ok. Let me make sure I understand your deployment more precisely.

You mentioned a [GHA action](https://github.com/drscotthawley/blog_quarto/actions/runs/3494794082), but also a local deployment. Did you show us logs of your local filesystem or GHA's?

**Comment by cscheid** [comment]
A followup question: In your repository's `main` branch, I see `55c0c37e` and `42b19423`. I see they were pushed with a message of `Built site for gh-pages`. Did you push these yourself? I'm trying to understand how those directories came to be.

**Comment by drscotthawley** [comment]
@cscheid  thanks for your response. 
 
All the logs I'm copy and pasting are from my Mac, so: local. I shared one link to a GitHub Actions page just to show that apparently that workflow finished.  

 I don't see the commits you mention when I look at main on GitHub. Here's a screenshot of what I see:  
 
<img width="1490" alt="Screen Shot 2022-11-18 at 8 47 26 AM" src="https://user-images.githubusercontent.com/13925685/202731817-c6c50ecf-3ddf-4b33-82d8-2af3e8223fbb.png">

I have manually pushed to main 3 times: 
1. Once for the initial adding of content to my repo (8 hours ago), and twice this morning:  
2. once after aborting the `quarto publish` process that was stalled for 7 hours, 
3. and once again after doing a `git checkout main` after seeing the "fatal" message in the output above.

**Comment by drscotthawley** [comment]
This morning, since I've been trying many different things to fix the broken blog, the `gh-pages` branch shows a number of commits in the past half hour: 
<img width="1508" alt="Screen Shot 2022-11-18 at 8 51 56 AM" src="https://user-images.githubusercontent.com/13925685/202732854-14ab81f5-6b0e-4add-b20b-032a957fc968.png">

**Comment by cscheid** [comment]
It's weird. I forked your repo, deleted those two directories, typed `quarto publish gh-pages` and got [this](https://cscheid.github.io/quarto-3368-repro/), so I can't currently reproduce. What version of quarto are you running?

**Comment by drscotthawley** [comment]
```
$ quarto --version
1.1.189
```

**Comment by drscotthawley** [comment]
Following your instructions & updating quarto didn't help. Here's what I did: 

```bash
$ cd ..
$ mv blog_quarto blog_quarto.O
$ git clone https://github.com/drscotthawley/blog_quarto.git
$ cd blog_quarto/
$ git rm -r 42b19423/ 55c0c37e/
$ git commit  [added message]
$ git push 
[download & install new version of quarto]
$ quarto --version
1.2.269
$ quarto publish
....
hangs at the same spot as before. 
```

**Comment by drscotthawley** [comment]
will reboot, just to see if that help....

**Comment by cscheid** [comment]
You're running a quite old version of `quarto`. Can you try one of our [stable releases](https://github.com/quarto-dev/quarto-cli/releases/tag/v1.2.269)?

**Comment by cscheid** [comment]
I missed the fact that you had updated on that shell transcript, I apologize.

**Comment by drscotthawley** [comment]
Rebooting caused the hanging to stop! :-) 

Still getting a `fatal: ... not a working tree` error, but the `quarto publish` process ran to completion:

```
origin	https://github.com/drscotthawley/blog_quarto (push)

To https://github.com/drscotthawley/blog_quarto
   6c2f3a0..24f097a  HEAD -> gh-pages
fatal: '25425f0b' is not a working tree

NOTE: GitHub Pages sites use caching so you might need to click the refresh
button within your web browser to see changes after deployment.

[✓] Published to https://github.com/drscotthawley/blog_quarto

NOTE: GitHub Pages deployments normally take a few minutes (your site updates
will be visible once the deploy completes)
```


**However**, there is still no website: https://drscotthawley.github.io/blog_quarto/ 404's.  :-( 


`gh-pages` [Deployment workflow](https://github.com/drscotthawley/blog_quarto/actions/runs/3498276034/jobs/5858424589) fails, may be due to a Jekyll error re use of the `cite` Liquid tag.  

Interesting that the website built for you, though @cscheid !  That should be independent of our individual setups. 🤔

**Comment by drscotthawley** [comment]
Ok, @cscheid this seems to be a MacOS / Apple Silicon issue, not GitHub, not Liquid.  In that:  

 I re-ran [the workflow above](https://github.com/quarto-dev/quarto-cli/issues/3368#issuecomment-1320137842) on an Ubuntu Linux system  (same version of `quarto` -- 1.2.269) and everything ran to completion and the blog is now deployed.

**Comment by drscotthawley** [comment]
....but not deployed correctly:  Only one post (out of about 30) is there. Even the main blog URL redirects to that one post. 😭

**Comment by cscheid** [comment]
@drscotthawley I'm on MacOS Monterey M1 Max :(

**Comment by cscheid** [comment]
> ....but not deployed correctly: Only one post (out of about 30) is there. Even the main blog URL automatically redirects to that one post. 😭

That is ok, that is because you don't actually have a page with listings. That's something you have to do, and it should be easy to fix!

**Comment by xiaoxl** [comment]
> Is your git repository on a non-standard filesystem, by any chance? eg a dropbox folder, google drive, onedrive

I have the same issue regularly and my repository is also in my google drive (I am using win11). Is this an issue?

**Comment by enzokro** [comment]
Hi all,

I was facing the same issue on a fresh Linux cloud machine. The publish command would hang and no changes went through. I also have a recent Quarto version: `1.2.269`

I was able to fix this (at least, get the changes to publish) with the following steps:

- Running `git worktree prune`. This allowed me to checkout the gh-pages branch. Else, I'd get the "fatal... already checkout out" error. 

After the checkout, my gh-pages branch had seriously diverged thanks to all of the `quarto publish` attempts. I had to run:  
`git pull --rebase`  
Because the branches even had a different history. 

Now, the branch was happy and I was able to push gh-pages to github!  

This prevented the hang and changes went through! Obviously this is an ugly bandage and there's a lot going on... hopefully it points others in a good direction.
